src/api/app/views/webui/cloud/upload_jobs/index.html.haml

```diff
@@ -15,7 +16,8 @@
       - @upload_jobs.each do |job|
         %tr
           %td= job.id
-          %td= job.filename
+          %td= link_to(job.filename, controller: '/webui/package', action: 'binary', project: job.project, package: job.package, repository: job.repository, arch: job.arch, filename: job.filename)
```

Comment:
I wouldn't link to the filename here but to the binaries page instead. The reason it is likely that the file does not exist anymore resulting in a 500. Images get quite frequently rebuild...

Reply:
If the binary got removed it also won't be in the list of binaries, right?
What's the point of linking to an empty page then?

I have to think more about it, but maybe the best solution is to check if the binary exists and then link it or not, what do you think?

Reply:
> If the binary got removed it also won't be in the list of binaries, right?
What's the point of linking to an empty page then?

It would contain the new build image / binary (the rebuild).

> I have to think more about it, but maybe the best solution is to check if the binary exists and then link it or not, what do you think?

This is not easily possible.

Reply:
> It would contain the new build image / binary (the rebuild).

I guess that if this is already created... or the binaries are wiped out whenever a new one is created?